mitreattack/navlayers/core/README.md

```diff
@@ -0,0 +1,245 @@
+
+# set a description
+layer_example.layer.description = "This is a demonstration of how to set up a layer piece by piece"
+
+# configure the "filters" object
+layer_example.layer.filters = dict(platforms=['example'])  # platforms can be provided during initialization
+layer_example.layer.filters.platforms = ['a', 'b', 'c']  # or separately
+
+# configure the 'sorting' setting
+layer_example.layer.sorting = 3  # 0: sort ascending alphabetically by technique name
+# 1: sort descending alphabetically by technique name
+# 2: sort ascending by technique score
+# 3: sort descending by technique score
+
+# configure the layout object
+layer_example.layer.layout = dict(layout="side",
+                                  showID=True,
+                                  showName=True,
+                                  showAggregateScores=True,
+                                  countUnscored=True,
+                                  aggregateFunction="sum")  # average, sum, max, min
+
+# configure whether or not to hide disabled techniques
+layer_example.layer.hideDisabled = True
+# configure the gradient object
+layer_example.layer.gradient = dict(minValue=-100, maxValue=100,
+                                    colors=["#DAF7A6", "#FFC300", "#FF5733", "#C70039", "#900C3F", "#581845"])
+# configure collection of legend items 
+layer_example.layer.legendItems = [dict(label='A', color='#DAF7A6'), dict(label='B', color='#581845')]
+# configure collection of metatdata values
+layer_example.layer.metadata = [dict(name='example metadata', value='This is an example')]
+# create listing of techniques in this layer
+layer_example.layer.techniques = [dict(techniqueID='T1000', tactic='privilege-escalation', score=15, color='#AABBCC'),
+                                  dict(techniqueID='T1000.1', tactic='privilege-escalation', score=1, comment='Demo')]
+```
+This first example utilizes the native dictionary form for initializing the layer. This approach is similar to the 
+method used by the automated import process, but may not be the most intuitive for users. An alternative method, 
+displayed below, is to create and modify instances of the core objects in the library. Please note, these two examples 
+produce equivalent internal layers once completed.
+
+```python
+import mitreattack.navlayers as navlayers
+
+layer_example = navlayers.Layer()
+layer_build = navlayers.core._LayerObj(name="example", domain="enterprise-attack") # arguments required for every layer
+
+# configure the versions object
+versions_obj = navlayers.Versions()
+versions_obj.layer = "4.2"
+versions_obj.attack = "9.1"
+versions_obj.navigator = "4.2"
+layer_build.versions = versions_obj
+
+# set a description
+layer_build.description = "This is a demonstration of how to set up a layer piece by piece"
+
+# configure the "filters" object
+filter_obj = navlayers.core.Filter(domain="enterprise-attack")
+filter_obj.platforms = ['a', 'b', 'c']
```

Comment:
Should probably use an actual set of platforms so users don't try to actually use the layer built from this example.

Reply:
Fair enough.